dotCMS/src/main/java/com/dotmarketing/beans/MultiTree.java

```diff
@@ -1,126 +1,176 @@
 package com.dotmarketing.beans;
 
-import java.io.Serializable;
-
 import com.dotcms.repackage.org.apache.commons.lang.builder.EqualsBuilder;
 import com.dotcms.repackage.org.apache.commons.lang.builder.HashCodeBuilder;
 import com.dotcms.repackage.org.apache.commons.lang.builder.ToStringBuilder;
 
+import java.io.Serializable;
+
 /**
  *
- * @author  maria
+ * @author maria
  */
 public class MultiTree implements Serializable {
 
     private static final long serialVersionUID = 1L;
 
-	/** identifier field */
-    private String parent1;
+    public static final String LEGACY_RELATION_TYPE="LEGACY_RELATION_TYPE";
+    
+    /** identifier field */
+    private String htmlPage;
 
     /** identifier field */
-    private String parent2;
+    private  String container;
 
     /** identifier field */
-    private String child;
+    private  String contentlet;
 
     /** nullable persistent field */
-    private String relationType;
+    private  String relationType;
 
     /** nullable persistent field */
-    private int treeOrder;
+    private  int treeOrder;
 
     /** full constructor */
 
-    public MultiTree(String parent1, String parent2, String child, java.lang.String relationType, int treeOrder) {
-
-        this.parent1 = parent1;
-        this.parent2 = parent2;
-        this.child = child;
-        this.relationType = relationType;
-        this.treeOrder = treeOrder;
+    public MultiTree(String htmlPage, String container, String child, String relationType, int treeOrder) {
+        this.htmlPage = htmlPage;
+        this.container = container;
+        this.contentlet = child;
+        this.relationType = (relationType==null) ? LEGACY_RELATION_TYPE : relationType;
+        this.treeOrder =  (treeOrder<0)? 0  : treeOrder;
     }
 
     /** default constructor */
     public MultiTree() {
-
-        this.parent1 = "";
-        this.parent2 = "";
-        this.child = "";
+        this(null, null, null, LEGACY_RELATION_TYPE,0);
     }
 
     /** minimal constructor */
 
-    public MultiTree(String parent1, String parent2,  String child) {
-        this.parent1 = parent1;
-        this.parent2 = parent2;
-        this.child = child;
+    public MultiTree(String htmlPage, String container, String child) {
+        this(htmlPage, container, child, LEGACY_RELATION_TYPE,0);
     }
-
+    /**
+     * 
+     * {@link #getContentlet()}
+     */
+    @Deprecated
     public String getChild() {
-        return this.child;
+        return  getContentlet();
+    }
+
+    public String getContentlet() {
+        return this.contentlet;
     }
 
-    public void setChild(String child) {
-        this.child = child;
+
+    @Deprecated
+    public MultiTree setChild(String child) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)